### PR TITLE
Add a flag for choosing trace endpoint type

### PR DESCRIPTION
**Pull request by prmsrswt**
Add a flag to specify if the trace endpoint is an agent or a collector.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -651,7 +657,9 @@ func parseFlags() (config, error) {
 	flag.StringVar(&cfg.internalTracing.serviceName, "internal.tracing.service-name", "observatorium_api",
 		"The service name to report to the tracing backend.")
 	flag.StringVar(&cfg.internalTracing.endpoint, "internal.tracing.endpoint", "",
-		"The full URL of the trace collector. If it's not set, tracing will be disabled.")
+		"The full URL of the trace agent or collector. If it's not set, tracing will be disabled.")
+	flag.StringVar(&cfg.internalTracing.endpointType, "internal.tracing.endpoint-type", tracing.EndpointTypeAgent,
+		"The tracing endpoint type. Options: 'agent', 'collector'.")
```
Comment (squat) [inline main.go:662]: nit: you may want to use the consts to template the string of available options

File: tracing/tracing.go [hunk 2]
```diff
@@ -11,17 +11,30 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-// InitTracer creates an OTel TracerProvider that exports the traces to a Jaeger collector.
-func InitTracer(serviceName, collectorEndpoint string, samplingFraction float64) (tp trace.TracerProvider, closer func(), err error) {
-	disabled := collectorEndpoint == ""
+const (
+	EndpointTypeCollector = "collector"
```
Comment (squat) [inline tracing/tracing.go:15]: nit: let's add comments for all exported types, including these consts

**Review by squat — APPROVED** [review]
Looks nice overall! Just some small nits

File: tracing/tracing.go [hunk 3]
```diff
@@ -11,17 +11,31 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-// InitTracer creates an OTel TracerProvider that exports the traces to a Jaeger collector.
-func InitTracer(serviceName, collectorEndpoint string, samplingFraction float64) (tp trace.TracerProvider, closer func(), err error) {
-	disabled := collectorEndpoint == ""
+// EndpointType represents the type of the tracing endpoint.
+const (
```
Comment (squat) [inline tracing/tracing.go:15]: I slightly nicer way to use Go's type system here is to declare a new string type and to make these consts specific instances of the new type, e.g.:
```golang
type EndpointType string
const (
    CollectorEndpointType EndpointType = "collector"
    AgentEndpointType EndpointType = "agent"
)

**Review by kakkoyun — APPROVED** [review]
💯 LGTM

File: README.md [hunk 4]
```diff
@@ -58,7 +58,9 @@ Usage of ./observatorium:
   -debug.name string
     	A name to add as a prefix to log lines. (default "observatorium")
   -internal.tracing.endpoint string
-    	The full URL of the trace collector. If it's not set, tracing will be disabled.
+    	The full URL of the trace agent or collector. If it's not set, tracing will be disabled.
+  -internal.tracing.endpoint-type string
```
Comment (bwplotka) [inline README.md:62]: Just curious why do we need this? What does it change?

Comment (prmsrswt) [reply README.md:62]: Agents and collector uses different protocols (IIRC Agent uses Thrift over UDP and collector uses HTTP).

**Pull request merged**